server/vulnerabilities/centos.go

```diff
@@ -0,0 +1,92 @@
+package vulnerabilities
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+
+	"github.com/fleetdm/fleet/v4/server/config"
+	"github.com/fleetdm/fleet/v4/server/fleet"
+	"github.com/fleetdm/fleet/v4/server/vulnerabilities/vuln_centos"
+	kitlog "github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+)
+
+// centosPostProcessing performs processing over the list of vulnerable rpm packages
+// and removes the vulnerabilities where the CVEs are known to be fixed.
+func centosPostProcessing(
+	ctx context.Context,
+	ds fleet.Datastore,
+	db *sql.DB,
+	logger kitlog.Logger,
+	config config.FleetConfig,
+) error {
+	centOSPkgs, err := vuln_centos.LoadCentOSFixedCVEs(ctx, db, logger)
+	if err != nil {
+		return fmt.Errorf("failed to fetch CentOS packages: %w", err)
+	}
+	level.Info(logger).Log("centosPackages", len(centOSPkgs))
+	if len(centOSPkgs) == 0 {
+		return nil
+	}
+
+	rpmVulnerable, err := ds.ListVulnerableSoftwareBySource(ctx, "rpm_packages")
+	if err != nil {
+		return fmt.Errorf("failed to list vulnerable software: %w", err)
+	}
+	level.Info(logger).Log("vulnerable rpm_packages", len(rpmVulnerable))
+	if len(rpmVulnerable) == 0 {
+		return nil
+	}
+
+	var fixedCVEs []fleet.SoftwareVulnerability
+	var softwareCount int
+	for _, software := range rpmVulnerable {
+		if software.Vendor != "CentOS" {
```

Comment:
Could we do this filtering in `ds.ListVulnerableSoftwareBySource` ?

Reply:
To keep things simple I'll keep it generic on this iteration, assuming most `rpm_packages` will have `CentOS` as vendor.

Let me know if it makes sense.

Reply:
Could easily be redhat, but not a huge deal.